library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Block.hs

```diff
@@ -1028,6 +1050,48 @@ generateIf ty cond tr fl = do
   setBlock ifExit
   phi ty [(t, ifThen), (f, ifElse)]
 
+-- | @generateSwitch@ is a layer over @switch@, similarly to how
+-- @generateIf@ is a layer over @cbr@.
+--
+-- @
+-- Block.generateSwitch ty term cases
+-- @
+generateSwitch ::
+  ( RetInstruction m,
+    HasState "blockCount" Int m,
+    HasState "names" Names m,
+    HasState "symTab" SymbolTable m
+  ) =>
+  Type ->
+  Operand ->
+  [Symbol] ->
+  [C.Constant] ->
+  [m Operand] ->
+  m Operand
+generateSwitch ty tag names values cases = do
+  internNames <- mapM addBlock $ map ("switch." <>) names
+  defaultBlock <- addBlock "switch-default"
+  exitBlock <- addBlock "switch-exit"
+  _ <- switch tag defaultBlock $ zip values internNames
+  phiPairs <-
+    mapM
+      ( \(n, c) -> do
+          setBlock n
+          op <- c
+          _ <- br exitBlock
+          n' <- getBlock
+          pure (op, n')
+      )
+      (zip internNames cases)
+  setBlock defaultBlock
+  abort
+  _unreachable <- br exitBlock
+  let errVal = AST.ConstantOperand $ C.Int {C.integerBits = 8, C.integerValue = 0}
+  castedErr <- bitCast errVal ty
+  defaultBlock <- getBlock
+  setBlock exitBlock
```

Comment:
I've added two functions that will help with this in a2868b2. Namely `withBlock` and `withNewBlock`. I think it would be a good idea to try to use them, when possible, instead of manually switching from/to blocks.